core/lib/types/src/aggregated_operations.rs

```diff
@@ -29,6 +31,7 @@ fn l1_batch_range_from_batches(
 pub struct L1BatchCommitOperation {
     pub last_committed_l1_batch: L1BatchWithMetadata,
     pub l1_batches: Vec<L1BatchWithMetadata>,
+    pub l1_batch_committer: Arc<dyn L1BatchCommitter>,
```

Comment:
is `L1BatchCommitter` a good name for such an abstraction? I would use something like `L1BatchCommitDataGenerator`

Reply:
The thing is, for me `L1BatchCommitter` sounds like something that actually commits blocks to some DA layer, whereas in fact it just generattes data

Reply:
Agreed! `L1BatchCommitDataGenerator` is definitively a proper name for the trait.